### ISSUE TITLE
[Windows 8] Notes is not correctly installed and throws the "add-on is not correctly signed" error in console

### ISSUE THREAD

**Issue by SoftVision-EmilPasca** [mozilla/notes#196]
**[Notes]:**
- I have tested this issue on multiple machines with Windows 8 x32/x64 and this issue was reproducible only on one of the machines with Windows 8 x64

**[Affected versions]:**
- Firefox 54.0 and above

**[Affected Platforms]:**
- Windows 8

**[Steps to reproduce]:**
1. Navigate to https://testpilot.firefox.com/experiments/notes page.
2. Click on "Install Test Pilot & Enable Notes".
3. Observe the "Notes" sidebar.

**[Expected result]:**
- "Notes" sidebar si displayed after enabling the experiment.

**[Actual results]:**
- "Notes" sidebar is not displayed.

[Additional Notes]:
- When clicking the "Show sidebars" toolbar button the "Notes" option is not available.
- In Nightly, when changing the "xpinstall.signatures.required" pref to false, the add-on is successfully installed.
- Attached a screen recording of the issue:
![screenrecording_notes](https://user-images.githubusercontent.com/20083658/29130759-7d24bf3c-7d33-11e7-9ca0-5e96c167ad03.gif)

**Comment by johngruen** [comment]
Let's keep this open for now, but this has not been super reproducable

**Comment by Natim** [comment]
@SoftVision-EmilPasca can you still reproduce this with latests version of notes/nightly?

**Comment by SoftVision-EmilPasca** [comment]
I have retested this on Windows 8 x64 with the latest Firefox Release(56.0) and the latest Nightly(58.0a1-20171009220104) and the issue is still reproducible.

`Content Security Policy: Ignoring ‘x-frame-options’ because of ‘frame-ancestors’ directive.  (unknown)
Content Security Policy: Directive ‘frame-src’ has been deprecated. Please use directive ‘child-src’ instead. (unknown)
Content Security Policy: The page’s settings blocked the loading of a resource at self (“style-src https://testpilot.firefox.com https://code.cdn.mozilla.net”). Source: height:0px;. notes
Content Security Policy: The page’s settings blocked the loading of a resource at self (“style-src https://testpilot.firefox.com https://code.cdn.mozilla.net”). notes
Got install event for unknown install 0  amWebAPI.js:55
Got install event for unknown install 0 amWebAPI.js:55
Got install event for unknown install 0 amWebAPI.js:55
1507644224079 addons.xpi WARN Add-on @testpilot-addon is missing bootstrap method install
1507644224095 addons.xpi WARN Failed to remove temporary file C:\Users\svuser\AppData\Local\Temp\tmp-j4p.xpi for addon https://testpilot.firefox.com/files/@testpilot-addon/latest: [Exception... "Component returned failure code: 0x80520015 (NS_ERROR_FILE_ACCESS_DENIED) [nsIFile.remove]"  nsresult: "0x80520015 (NS_ERROR_FILE_ACCESS_DENIED)"  location: "JS frame :: resource://gre/modules/addons/XPIInstall.jsm :: removeTemporaryFile :: line 1596"  data: no] Stack trace: removeTemporaryFile()@resource://gre/modules/addons/XPIInstall.jsm:1596 < startInstall/<()@resource://gre/modules/addons/XPIInstall.jsm:1964
Got install event for unknown install 0  amWebAPI.js:55
Sending message that cannot be cloned. Are you trying to send an XPCOM object?  addonManager.js:206:10
[Exception... "Component returned failure code: 0x8000ffff (NS_ERROR_UNEXPECTED) [nsIPrefBranch.getCharPref]"  nsresult: "0x8000ffff (NS_ERROR_UNEXPECTED)"  location: "JS frame :: resource://gre/modules/addons/XPIProvider.jsm -> jar:file:///C:/Users/svuser/AppData/Roaming/Mozilla/Firefox/Profiles/3n5q6l2v.notes/extensions/@testpilot-addon.xpi!/bootstrap.js :: sendInitialPrefs :: line 753"  data: no]  (unknown)
1507644225631 addons.xpi WARN Add-on notes@mozilla.com is not correctly signed.
Sending message that cannot be cloned. Are you trying to send an XPCOM object?  addonManager.js:206:10
Sending message that cannot be cloned. Are you trying to send an XPCOM object?  addonManager.js:206:10
Sending message that cannot be cloned. Are you trying to send an XPCOM object? addonManager.js:206:10
Sending message that cannot be cloned. Are you trying to send an XPCOM object?  addonManager.js:206:10
Sending message that cannot be cloned. Are you trying to send an XPCOM object?  addonManager.js:206:10
Use of getAttributeNode() is deprecated. Use getAttribute() instead. disco-9ace9d9e22f3f62884f6.js:1:667243
Content Security Policy: The page’s settings blocked the loading of a resource at self (“style-src https://addons-discovery.cdn.mozilla.net 'sha256-DiZjxuHvKi7pvUQCxCVyk1kAFJEUWe+jf6HWMI5agj4='”).`

**Comment by Natim** [comment]
I think we should file a bugzilla issue for that.

**Comment by SoftVision-EmilPasca** [comment]
Refs : [bugzilla 1407268](https://bugzilla.mozilla.org/show_bug.cgi?id=1407268).

**Comment by Natim** [comment]
@ulfr tldr:

1. The notes@mozilla.com add-on is signed with the TestPilot experiment key
2. For some reasons the signature is invalid when using Windows 8 64bits with Firefox 32 bits installed in it
3. We checked and it is the case only with that TestPilot add-on
4. Even when we realised the last version of it lately it is still reproducible

Any ideas why this particular add-on with that particular setup would fail?
